zipline/finance/risk/cumulative.py

```diff
@@ -462,16 +462,17 @@ def calculate_beta(self):
 
         http://en.wikipedia.org/wiki/Beta_(finance)
         """
-        # it doesn't make much sense to calculate beta for less than two days,
-        # so return none.
-        if len(self.annualized_mean_returns) < 2:
-            return 0.0
 
         # Drop nans if there are gaps in the data
         algorithm_returns = self.algorithm_returns.dropna()
         benchmark_returns = \
             self.benchmark_returns.loc[algorithm_returns.index]
 
+        # it doesn't make much sense to calculate beta for less than two days,
+        # so return none.
+        if len(algorithm_returns) < 2:
+            return 0.0
```

Comment:
Talked with @ehebert about this earlier. The warning is returned when we try to calculate covariance between the benchmark and algo when there is only one data point of returns. len(self.annualized_mean_returns) <2: return 0.0 is meant to address that case. However, this length is calculated before NaNs are dropped from algorithm_returns and benchmark_returns